File: case_studies/tuto/skewing/refactor_if_in_loop/refactor_if_in_loop.cpp
#include <optitrust.h>
#include <stdlib.h>
int main() {
  const int N = 5000;
  for (int i = 0; i < N; i++) {
    if (i >= 10 && i < 15) {
      i++;
    }
  }
}
int main2() {
  const int N = 5000;
  for (int i = 0; i < N; i++) {
    if (10 <= i && i < 15) {
      i++;
    }
  }
}
int main3() {
  const int N = 5000;
  for (int i = 0; i < N; i++) {
    if (10 <= i && i < 15 && i < 18) {
      i++;
    }
  }
}

File: tests/loop/refactor_if_in_loop/refactor_if_in_loop.ml
open Optitrust
open Prelude

let _ =
  Run.script_cpp (fun _ ->
      !!Loop.refactor_if_in_loop [ cFunBody "main"; cFor "i" ];
      !!Loop.refactor_if_in_loop [ cFunBody "main3"; cFor "i" ];
      !!Loop.refactor_if_in_loop [ cFunBody "main4"; cFor "i" ];
      !!Loop.refactor_if_in_loop [ cFunBody "main2"; cFor "i" ])
